contrib/init/upstart/docker.conf

```diff
@@ -70,3 +70,25 @@ post-start script
 		echo "$DOCKER_SOCKET is up"
 	fi
 end script
+
+post-stop script
+        SHUTDOWN_RUNLEVELS='06'
+        if runlevel | cut -d' ' -f2 | grep -q [$SHUTDOWN_RUNLEVELS]; then
+                for container in $(docker-runc list -q); do
+                        docker-runc kill "$container" || true
```

Comment:
There's no standard/efficient way to determine that a received `SIGTERM` is due to a system reboot/shutdown AFAIK.

Maybe adding a `shutdown` command or similar to `docker` that init scripts can call when the system is about to go down?